templates/common/gcp/files/usr-local-bin-update-dns-server.yaml

```diff
@@ -0,0 +1,22 @@
+mode: 0755
+path: "/usr/local/bin/update-dns-server"
+contents:
+  inline: |
+    #!/bin/bash
+    # This is test code
+    
+    echo "Executing update-dns-server"
+
+    if ![-f /etc/NetworkManager/conf.d]; then
+      mkdir /etc/NetworkManager/conf.d
+      touch /etc/NetworkManager/conf.d/dns-servers.conf
+    fi
+    if ![-f /etc/NetworkManager/conf.d/dns-servers.conf]; then
+        touch /etc/NetworkManager/conf.d/dns-servers.conf
+    fi
+
+    echo "Added by OpenShift" >> /etc/NetworkManager/conf.d/dns-servers.conf
+    echo "[global-dns-domain-*]" >> /etc/NetworkManager/conf.d/dns-servers.conf
+    echo "servers=$(ip route get 8.8.8.8 | head -1 | cut -d' ' -f7),169.254.169.254" >> /etc/NetworkManager/conf.d/dns-servers.conf
```

Comment:
If you're changing DNS settings after NM is running you probably need to restart/reload it to pick up the new values. We've also used "nmcli general reload dns-rc" to force reconfiguration of DNS stuff, but I'm not positive that will do what you need here. It may be less invasive than a complete service restart though.

Reply:
I'll start with "nmcli general reload dns-rc" and update based on testing.